astropy/coordinates/tests/test_arrays.py

```diff
@@ -100,29 +100,28 @@ def test_array_coordinates_creation():
     """
     Test creating coordinates from arrays.
     """
-    from .. import Angle
-    from .. import ICRS, CartesianRepresentation
+    from .. import Angle, SkyCoord, CartesianRepresentation
 
-    c = ICRS(np.array([1, 2])*u.deg, np.array([3, 4])*u.deg)
+    c = SkyCoord(np.array([1, 2])*u.deg, np.array([3, 4])*u.deg)
     assert not c.ra.isscalar
```

Comment:
This change and the next 3 `ICRS`->`SkyCoord` changes are unnecessary.  And in fact, they might be counter-productive on the assumption that the parsing pathways for the frame classes and `SkyCoord` *might* be different.  

Maybe the thing to do is to leave this change in place but then add another test function(`test_array_frame_class_creation`)  which included everything from line 99 to 115, but using `ICRS` instead of `SkyCoord`? That way we get both separately tested.